pkg/clients/rds/rds.go

```diff
@@ -452,7 +458,59 @@ func IsUpToDate(p v1beta1.RDSInstanceParameters, db rds.DBInstance) (bool, error
 		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "Tags"),
 		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "SkipFinalSnapshotBeforeDeletion"),
 		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "FinalDBSnapshotIdentifier"),
-	), nil
+		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "ApplyModificationsImmediately"),
+		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "AllowMajorVersionUpgrade"),
+		cmpopts.IgnoreFields(v1beta1.RDSInstanceParameters{}, "MasterPasswordSecretRef"),
+	) && !pwdChanged, nil
+}
+
+// GetPassword fetches the referenced input password for an RDSInstance CRD and determines whether it has changed or not
+func GetPassword(ctx context.Context, kube *client.Client, inputCreds *v1alpha1.SecretKeySelector, outCreds *v1alpha1.SecretReference) (string, bool, error) {
+	if kube == nil {
+		return "", false, nil
+	}
+	if inputCreds == nil && outCreds == nil {
+		return "", false, nil
+	}
+
+	var err error
+	var newPwd string
+	var pwdChanged bool
+	if inputCreds != nil {
+		nn := types.NamespacedName{
+			Name:      inputCreds.Name,
+			Namespace: inputCreds.Namespace,
+		}
+		newPwd, err = getPasswordFromSecret(ctx, kube, nn, inputCreds.Key)
+		if err != nil {
+			return "", false, errors.Wrap(err, errGetPasswordSecretFailed)
+		}
+	}
+
+	if outCreds != nil {
+		nn := types.NamespacedName{
+			Name:      outCreds.Name,
+			Namespace: outCreds.Namespace,
+		}
+		outPwd, _ := getPasswordFromSecret(ctx, kube, nn, v1alpha1.ResourceCredentialsSecretPasswordKey)
+		if outPwd == newPwd {
+			return newPwd, false, nil
+		}
+	}
+
+	return newPwd, pwdChanged, nil
+}
+
+func getPasswordFromSecret(ctx context.Context, kube *client.Client, namespacedName types.NamespacedName, key string) (string, error) {
```

Comment:
From what I can tell, once we use `client.Client`, this function would not be really necessary and we can do the same operation in place in `GetPassword`.